src/clusterfuzz/_internal/build_management/build_archive.py

```diff
@@ -328,3 +331,34 @@ def open(archive_path: str) -> BuildArchive:
   if reader.file_exists(args_gn_path):
     return ChromeBuildArchive(reader)
   return DefaultBuildArchive(reader)
+
+
+def open_uri(uri: str) -> BuildArchive:
+  """Opens a build archive over HTTP. This is only compatible with chromium as
+  of now.
+
+  Args:
+      uri: the gs:// storage URI.
+
+  Returns:
+      the build archive.
+  """
+  assert unzip_over_http_compatible(uri)
+  match = CHROMIUM_GS_MATCHER.match(uri)
+  uri = f'https://commondatastorage.googleapis.com/{match.group(1)}'
```

Comment:
I think there might be a helper for this `gs://` to `https://` URL conversion somewhere in CF code already